torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1408,6 +1408,28 @@ at::Tensor& AtenXlaType::relu_(at::Tensor& self) const {
   return self;
 }
 
+at::Tensor AtenXlaType::hardtanh(const at::Tensor& self, at::Scalar min_val,
+                                 at::Scalar max_val) const {
+  return bridge::AtenFromXlaTensor(
+      XLATensor::clamp(bridge::GetXlaTensor(self), min_val, max_val));
```

Comment:
I have not looked at hardtanh, but isn't:
```
clamp(min, max, tanh(input))
```
Instead of:
```
clamp(min, max, input)
```

Reply:
No, I've checked both documentation and code. There's no tanh involved, it's linear between min and max.